include/nbl/builtin/hlsl/matrix_utils/transformation_matrix_utils.hlsl

```diff
@@ -19,82 +49,114 @@ matrix<T, 4, 4> getMatrix3x4As4x4(const matrix<T, 3, 4>& mat)
 	return output;
 }
 
-// TODO: use portable_float when merged
-//! multiplies matrices a and b, 3x4 matrices are treated as 4x4 matrices with 4th row set to (0, 0, 0 ,1)
-template<typename T>
-inline matrix<T, 3, 4> concatenateBFollowedByA(const matrix<T, 3, 4>& a, const matrix<T, 3, 4>& b)
+template<typename T, uint32_t N>
+inline matrix<T, 3, 3> getSub3x3(const matrix<T, N, 4>& mat)
 {
-	const auto a4x4 = getMatrix3x4As4x4(a);
-	const auto b4x4 = getMatrix3x4As4x4(b);
-	return matrix<T, 3, 4>(mul(a4x4, b4x4));
+	return matrix<T, 3, 3>(mat);
 }
 
-// /Arek: glm:: for normalize till dot product is fixed (ambiguity with glm namespace + linker issues)
-
-template<typename T>
-inline matrix<T, 3, 4> buildCameraLookAtMatrixLH(
-	const vector<T, 3>& position,
-	const vector<T, 3>& target,
-	const vector<T, 3>& upVector)
+template<uint32_t N, uint32_t M>
+inline matrix<float64_t, N, M> getAs64BitPrecisionMatrix(const matrix<float32_t, N, M>& mat)
 {
-	const vector<T, 3> zaxis = glm::normalize(target - position);
-	const vector<T, 3> xaxis = glm::normalize(hlsl::cross(upVector, zaxis));
-	const vector<T, 3> yaxis = hlsl::cross(zaxis, xaxis);
+	matrix<float64_t, N, M> output;
+	for (int i = 0; i < N; ++i)
+		output[i] = mat[i];
 
-	matrix<T, 3, 4> r;
-	r[0] = vector<T, 4>(xaxis, -hlsl::dot(xaxis, position));
-	r[1] = vector<T, 4>(yaxis, -hlsl::dot(yaxis, position));
-	r[2] = vector<T, 4>(zaxis, -hlsl::dot(zaxis, position));
+	return output;
+}
 
-	return r;
+namespace transformation_matrix_utils_impl
+{
+	template<typename T>
+	inline T determinant_helper(const matrix<T, 3, 3>& mat, vector<T, 3>& r1crossr2)
+	{
+		r1crossr2 = hlsl::cross(mat[1], mat[2]);
+		return hlsl::dot(mat[0], r1crossr2);
+	}
 }
 
-template<typename T>
-inline matrix<T, 3, 4> buildCameraLookAtMatrixRH(
-	const vector<T, 3>& position,
-	const vector<T, 3>& target,
-	const vector<T, 3>& upVector)
+template<typename T, uint32_t N, uint32_t M>
+inline matrix<T, 3, 3> getSub3x3TransposeCofactors(const matrix<T, N, M>& mat)
```

Comment:
Please document it returns [adjugate matrix](<https://en.wikipedia.org/wiki/Adjugate_matrix>) of `mat`